src/sentry/api/endpoints/organization_member_invite/index.py

```diff
@@ -0,0 +1,212 @@
+        # not, then POST creates an invite request, not an invite.
+        "POST": ["member:read", "member:write", "member:admin", "member:invite"],
+    }
+
+
+class MemberInviteAndStaffPermission(StaffPermissionMixin, MemberInvitePermission):
+    pass
+
+
+def _can_invite_member(
+    request: Request,
+    organization: Organization | RpcOrganization | RpcUserOrganizationContext,
+) -> bool:
+    scopes = request.access.scopes
+    is_role_above_member = "member:admin" in scopes or "member:write" in scopes
+    if isinstance(organization, RpcUserOrganizationContext):
+        organization = organization.organization
+
+    if is_role_above_member:
+        return True
+    if "member:invite" not in scopes:
+        return False
+    return not organization.flags.disable_member_invite
+
+
+def _create_invite_object(
+    request, organization, result, is_request: bool
+) -> OrganizationMemberInvite:
+    with transaction.atomic(router.db_for_write(OrganizationMemberInvite)):
+        teams = []
+        for team in result.get("teams", []):
+            teams.append({"id": team.id, "slug": team.slug, "role": None})
+
+        omi = OrganizationMemberInvite(
+            organization=organization,
+            email=result["email"],
+            role=result["orgRole"],
+            inviter_id=request.user.id,
+            organization_member_team_data=teams,
+            invite_status=(
+                InviteStatus.REQUESTED_TO_BE_INVITED.value
+                if is_request
+                else InviteStatus.APPROVED.value
+            ),
+        )
+
+        omi.save()
+
+    create_audit_entry(
+        request=request,
+        organization_id=organization.id,
+        target_object=omi.id,
+        data=omi.get_audit_log_data(),
+        event=(
+            (audit_log.get_event_id("INVITE_REQUEST_ADD"))
+            if is_request
+            else (audit_log.get_event_id("MEMBER_INVITE"))
+        ),
+    )
+    return omi
+
+
+@region_silo_endpoint
+@extend_schema(tags=["Organizations"])
+class OrganizationMemberInviteIndexEndpoint(OrganizationEndpoint):
+    # TODO (mifu67): make these PUBLIC once ready
+    publish_status = {
+        "GET": ApiPublishStatus.EXPERIMENTAL,
+        "POST": ApiPublishStatus.EXPERIMENTAL,
+    }
+    permission_classes = (MemberInviteAndStaffPermission,)
+    owner = ApiOwner.ENTERPRISE
+
+    def _invite_member(self, request, organization) -> Response:
+        allowed_roles = get_allowed_org_roles(request, organization, creating_org_invite=True)
+
+        is_member = not request.access.has_scope("member:admin") and (
+            request.access.has_scope("member:invite")
+        )
+
+        validator = OrganizationMemberInviteRequestValidator(
+            data=request.data,
+            context={
+                "organization": organization,
+                "allowed_roles": allowed_roles,
+                "allow_retired_roles": not features.has("organizations:team-roles", organization),
```

Comment:
nit: could remove `allow_retired_roles` from `context` and simply do the feature flag check where this is used inside the validator since we're already passing `organization` (it's only used in `validate_orgRole`)